tfhe/src/integer/bigint/u512.rs

```diff
@@ -1,313 +1,6 @@
-
-    fn shl(mut self, rhs: u32) -> Self::Output {
-        self <<= rhs;
-        self
-    }
-}
-
-impl std::ops::ShlAssign<u32> for U512 {
-    fn shl_assign(&mut self, shift: u32) {
-        super::algorithms::shl_assign(self.0.as_mut_slice(), shift);
-    }
-}
-
-impl std::ops::Shl<usize> for U512 {
-    type Output = Self;
-
-    fn shl(mut self, rhs: usize) -> Self::Output {
-        self <<= rhs;
-        self
-    }
-}
-
-impl std::ops::ShrAssign<usize> for U512 {
-    fn shr_assign(&mut self, shift: usize) {
-        super::algorithms::shr_assign(self.0.as_mut_slice(), shift as u32);
-    }
-}
-
-impl std::ops::Shr<usize> for U512 {
-    type Output = Self;
-
-    fn shr(mut self, rhs: usize) -> Self::Output {
-        self >>= rhs;
-        self
-    }
-}
-
-impl std::ops::ShlAssign<usize> for U512 {
-    fn shl_assign(&mut self, shift: usize) {
-        super::algorithms::shl_assign(self.0.as_mut_slice(), shift as u32);
-    }
-}
-
-impl std::ops::Not for U512 {
-    type Output = Self;
-
-    fn not(mut self) -> Self::Output {
-        super::algorithms::bitnot_assign(self.0.as_mut_slice());
-        self
-    }
-}
-
-impl std::ops::BitAnd<Self> for U512 {
-    type Output = Self;
-
-    fn bitand(mut self, rhs: Self) -> Self::Output {
-        self &= rhs;
-        self
-    }
-}
-
-impl std::ops::BitAndAssign<Self> for U512 {
-    fn bitand_assign(&mut self, rhs: Self) {
-        super::algorithms::bitand_assign(self.0.as_mut_slice(), rhs.0.as_slice())
-    }
-}
-
-impl std::ops::BitOrAssign<Self> for U512 {
-    fn bitor_assign(&mut self, rhs: Self) {
-        super::algorithms::bitor_assign(self.0.as_mut_slice(), rhs.0.as_slice())
-    }
-}
-
-impl std::ops::BitOr<Self> for U512 {
-    type Output = Self;
-
-    fn bitor(mut self, rhs: Self) -> Self::Output {
-        self |= rhs;
-        self
-    }
-}
-
-impl std::ops::BitXorAssign<Self> for U512 {
-    fn bitxor_assign(&mut self, rhs: Self) {
-        super::algorithms::bitxor_assign(self.0.as_mut_slice(), rhs.0.as_slice())
-    }
-}
-
-impl std::ops::BitXor<Self> for U512 {
-    type Output = Self;
-
-    fn bitxor(mut self, rhs: Self) -> Self::Output {
-        self ^= rhs;
-        self
-    }
-}
-
-impl From<u8> for U512 {
-    fn from(value: u8) -> Self {
-        Self::from(value as u128)
-    }
-}
-
-impl From<u16> for U512 {
-    fn from(value: u16) -> Self {
-        Self::from(value as u128)
-    }
-}
-
-impl From<u32> for U512 {
-    fn from(value: u32) -> Self {
-        Self::from(value as u128)
-    }
-}
-
-impl From<u64> for U512 {
-    fn from(value: u64) -> Self {
-        Self::from(value as u128)
-    }
-}
-
-impl From<u128> for U512 {
-    fn from(value: u128) -> Self {
-        Self([
-            (value & u128::from(u64::MAX)) as u64,
-            (value >> 64) as u64,
-            0,
-            0,
-            0,
-            0,
-            0,
-            0,
-        ])
-    }
-}
+pub type U512 = super::static_unsigned::StaticUnsignedBigInt<8>;
```

Comment:
looks odd to use the number of u64 blocks in the generic, let's use the number of bits ? normally computations on const generics params are possible in curly brace IIRC

Reply:
We can't 
neither 
```rust
struct Lol<const B: usize> {
    data: [u64; {B / u64::BITS as usize}]
}
```

nor
```
struct Lol<const B: usize> {
    data: [u64; B / u64::BITS as usize]
}
```

Const generics in rust are still lacking a bit, we would also need a kind of `static_assert(B % u64::BITS as usize == 0)` to avoid mistake such as `U119 = StaticUnsignedBigInt<119>`

Reply:
ok got it, too bad